src/lz4.rs

```diff
@@ -229,18 +230,20 @@ pub mod lz4 {
                     _ => BlockMode::Linked,
                 })
                 .build(Cursor::new(vec![]))?;
-            Ok(Self { inner: Some(inner) })
+            Ok(Self {
+                inner: Mutex::new(Some(inner)),
+            })
         }
 
         /// Compress input into the current compressor's stream.
         pub fn compress(&mut self, input: &[u8]) -> PyResult<usize> {
-            crate::io::stream_compress(&mut self.inner, input)
+            crate::io::stream_compress(&mut self.inner.lock().unwrap(), input)
```

Comment:
If it eases things, I'm not adverse to adding [`parking_lot::Mutex`](https://docs.rs/parking_lot/latest/parking_lot/type.Mutex.html)